### PR TITLE
Set up issue template

### PR DESCRIPTION
**Pull request by kberzinch**

### CODE REVIEW

File: ISSUE_TEMPLATE.md
Comment (kberzinch) [inline ISSUE_TEMPLATE.md]: So we can actually just have them request a review from the Software Instructors team if we want...idk

File: ISSUE_TEMPLATE.md
Comment (joshhting) [inline ISSUE_TEMPLATE.md]: I prefer "branch" over "parallel version" because parallel version could just as easily refer to forks

File: ISSUE_TEMPLATE.md
Comment (sastorer) [reply ISSUE_TEMPLATE.md]: yes do that....that'll be easier than them asking us for out github usernames

**Review by ebarnette — APPROVED** [review]
lgtm

File: ISSUE_TEMPLATE.md
Comment (jgkamat) [reply ISSUE_TEMPLATE.md]: You can't request people for review if you don't have read permissions already.

File: ISSUE_TEMPLATE.md
Comment (kberzinch) [reply ISSUE_TEMPLATE.md]: Agreed

Comment (sastorer) [reply ISSUE_TEMPLATE.md]: I was just trying to omit as many keywords as possible
I'm cool with going back to branch

File: ISSUE_TEMPLATE.md
Comment (JasonGibson274) [inline ISSUE_TEMPLATE.md]: Can we say fork here?

File: ISSUE_TEMPLATE.md
Comment (JasonGibson274) [inline ISSUE_TEMPLATE.md]: The window size is 30 characters tall and 80 long

File: ISSUE_TEMPLATE.md
Comment (jgkamat) [inline ISSUE_TEMPLATE.md]: Are you trying to avoid saying 'fork' here? Seems better to use the proper terminology.

File: ISSUE_TEMPLATE.md
Comment (kberzinch) [reply ISSUE_TEMPLATE.md]: Okay, so that will just have to be an IRL thing.

File: ISSUE_TEMPLATE.md
Comment (jgkamat) [inline ISSUE_TEMPLATE.md]: Are you trying to avoid saying 'clone' here? Seems better to use the proper terminology.

File: ISSUE_TEMPLATE.md
Comment (jgkamat) [inline ISSUE_TEMPLATE.md]: Where should this branch branch off from?

File: ISSUE_TEMPLATE.md
Comment (joshhting) [inline ISSUE_TEMPLATE.md]: Say something like "Put something like 'fix #xyz' in the pr description where xyz is your issue #".

**Review by ebarnette — APPROVED** [review]

**Review by joshhting — CHANGES_REQUESTED** [review]
Also add an objective to copy paste this stuff into their issue, and name the issue after their gt username so they don't get them confused with other issues

File: ISSUE_TEMPLATE.md
Comment (jgkamat) [inline ISSUE_TEMPLATE.md]: Where should `yourgtusername/merge-conflict` branch off from? If you do it from the wrong place you won't get a merge conflict.

File: ISSUE_TEMPLATE.md
Comment (sastorer) [reply ISSUE_TEMPLATE.md]: I think I want to leave that up to them to figure out.....it will help them understand the difference between fork/clone/etc.

File: ISSUE_TEMPLATE.md
Comment (jgkamat) [inline ISSUE_TEMPLATE.md]: Are you trying to avoid saying merge, cherry pick, or rebase? This is incredibly vauge.

**Comment by kberzinch** [comment]
@joshhting once this is merged, this file will be autofilled for all new issues on this repo, so we don't need them to copy-paste anything.

File: ISSUE_TEMPLATE.md
Comment (jgkamat) [inline ISSUE_TEMPLATE.md]: does yourgtusername refer to branch or remote?

**Review by jgkamat — CHANGES_REQUESTED** [review]

File: ISSUE_TEMPLATE.md
Comment (joshhting) [reply ISSUE_TEMPLATE.md]: I guess specifying we're branching off of master wouldn't hurt

File: ISSUE_TEMPLATE.md
Comment (kberzinch) [reply ISSUE_TEMPLATE.md]: https://github.com/RoboJackets/jr/pull/2#discussion_r149197079

File: ISSUE_TEMPLATE.md
Comment (sastorer) [reply ISSUE_TEMPLATE.md]: @kberzinch yeah I would add the 80 char per line limit here in the description

File: ISSUE_TEMPLATE.md
Comment (kberzinch) [reply ISSUE_TEMPLATE.md]: While I see your point, I'm not sure this is phrased clearly enough for people to figure it out...

File: ISSUE_TEMPLATE.md
Comment (joshhting) [reply ISSUE_TEMPLATE.md]: I think we can leave that as an exercise to the trainee to ensure a merge conflict happens

File: ISSUE_TEMPLATE.md
Comment (kberzinch) [reply ISSUE_TEMPLATE.md]: Per person?

File: ISSUE_TEMPLATE.md
Comment (sastorer) [reply ISSUE_TEMPLATE.md]: any of those work....it's up to interpretation

File: ISSUE_TEMPLATE.md
Comment (jgkamat) [reply ISSUE_TEMPLATE.md]: I don't think that obsfucating the terminology helps them understand anything, fwiw.

Maybe you could just remove these steps completely, which would be a much more realistic scenario.

File: ISSUE_TEMPLATE.md
Comment (sastorer) [reply ISSUE_TEMPLATE.md]: yeah

File: ISSUE_TEMPLATE.md
Comment (sastorer) [reply ISSUE_TEMPLATE.md]: Honestly I'm not sure. @joshhting ?

File: ISSUE_TEMPLATE.md
Comment (sastorer) [reply ISSUE_TEMPLATE.md]: Ideally we'd want them to look it up or rationalize it out themselves if they don't understand...we don't want them to just type words into the terminal from this list

File: ISSUE_TEMPLATE.md
Comment (joshhting) [reply ISSUE_TEMPLATE.md]: We should specify branch here

File: ISSUE_TEMPLATE.md
Comment (joshhting) [reply ISSUE_TEMPLATE.md]: We haven't taught them cherry pick or rebase yet, right? To them the only thing they that makes sense here is merging

File: ISSUE_TEMPLATE.md
Comment (jgkamat) [inline ISSUE_TEMPLATE.md]: "fork’s gtusername remote branch" is redundant and weird.

How about "gtusername branch on your fork"

File: ISSUE_TEMPLATE.md
Comment (JasonGibson274) [reply ISSUE_TEMPLATE.md]: yeah that is the limit per person. Anything outside of it just will not be shown.

File: ISSUE_TEMPLATE.md
Comment (jgkamat) [reply ISSUE_TEMPLATE.md]: I think that there are multiple issues here:

1. The entire sentence is confusing. The actual intent of the sentence is "Fork the repo", but somehow it manages to be >200 chars.
2. The intent of the sentence is conflicting. The overall feel of the sentence feels like we're trying to hide information, but the latter half of the sentence feels like we're trying to help them out (by providing a very long description).

IMHO you should either:

Make it "easy": "Fork the repo on GitHub. [OPTIONAL explanation]"
Make it "hard": "Copy the repo on GitHub"
Make it realistic: omit this statement completely

Seeing as this step is required for literally everything else, I like "easy" the most, but I can see the merit of making it "realistic". I don't particularly (still) like "hard".

File: ISSUE_TEMPLATE.md
Comment (jgkamat) [reply ISSUE_TEMPLATE.md]: fwiw, `git-apply` is the command that I originally though you wanted us to use (since you say "apply the changes"), but it dosen't make sense in this context, unless we want them to format patches.

**Comment by kberzinch** [comment]
@jgkamat I committed some changes _right_ after your review that clobbered both - I think I cleared up both of them though.

**Comment by kberzinch** [comment]
I'm going to do a style pass, there's a couple things that are just messy

**Review by joshhting — APPROVED** [review]

**Comment by kberzinch** [comment]
@jgkamat you still have a blocking review - are there any other changes you want to discuss?

File: ISSUE_TEMPLATE.md
Comment (jgkamat) [inline ISSUE_TEMPLATE.md]: I think that this should be easy to understand or removed, see comments on fork above.

**Review by jgkamat — COMMENTED** [review]
Overall, I don't like the wording too much, because it's very vauge (it's hard to understand what to do at each step, even when you already know what's trying to be taught), but I suppose we're out of time so we should go for it anyway.

**Review by jgkamat — APPROVED** [review]

**Pull request merged**
